### PR TITLE
Complete port to React

### PR DESCRIPTION
**Pull request by arutr**
This should include all pages of the original EJS version ported into React, as well as functioning endpoints.
A new API has been created to avoid constant copying-and-pasting of same code for interacting with the database.
I have also created a new, more relevant README and tidied up the root directory.

The only issue I am aware of is that resource forecasting does not seem to function at all. Team search still does not work.
After transferring to the new repository, I will focus on fixing these issues.

### CODE REVIEW

File: client/src/api.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(body)
+    }).then(response => response.json())
+        .then(redirectHandler)
+        .catch(console.log);
+};
+
+exports.handleInputChange = function(event) {
```
Comment (gwolverson) [inline client/src/api.js:19]: could use arrow functions here as well;

exports.handleInputChange = (event) => {
...
}

File: client/src/components/Project/ChangeItems.js [hunk 2]
```diff
@@ -3,30 +3,37 @@
...
                     <tbody>
-                    <tr>
-                        <td>
-                            <Button href="/project/:projectCode/:changeItem" bsStyle="success">View</Button>
-                        </td>
-                        <td>Placeholder</td>
-                        <td>Placeholder</td>
-                        <td>DD/MM/YYYY</td>
-                    </tr>
+                    {this.props.changeItems.map(changeItem => {
```
Comment (gwolverson) [inline client/src/components/Project/ChangeItems.js]: This should be seperate out into it's own component. Should nest components in React, best to keep it all modular

File: client/src/components/Resource/ResourceForecast.js [hunk 3]
```diff
@@ -15,96 +18,116 @@ class Forecast extends Component {
...
-                            <a href="/project/:projectCode/:changeItem/:requiredResource">
-                                Placeholder
-                            </a>
-                        </td>
-                        <td>Placeholder</td>
-                    </tr>
+                    {this.props.projects.forEach(project => {
+                        project.changeItems.forEach(changeItem => {
+                            totalDays += changeItem.resourcesRequired.impact.days;
+                            return (
```
Comment (gwolverson) [inline client/src/components/Resource/ResourceForecast.js]: Again - this 'return' section should be separated out into it's own component and rendered that way

**Review by gwolverson — CHANGES_REQUESTED** [review]
Good work, only a few changes to address

File: client/src/api.js (hunk 1 above)
Comment (arutr) [reply client/src/api.js:19]: Unfortunately due to arrow function's nature, `this` cannot be bound to any other object. Same goes for other functions that use `this`.

Comment (gwolverson) [reply client/src/api.js:19]: Ah yeah good point

**Pull request merged**